src/components/HomePage/SearchResultsArea/Offer/OfferInteractionOptions.js

```diff
@@ -72,9 +91,11 @@ const OfferVisibilityOptions = ({
                     }
                     {
                         (
-                            visibilityState.isVisible ||
-                            !visibilityState.isDisabled ||
-                            sessionData?.company?._id === offer.owner
+                            (visibilityState.isVisible ||
+                                !visibilityState.isDisabled ||
+                                sessionData?.company?._id === offer.owner) &&
+                            (sessionData?.company?._id === offer.owner ||
+                                sessionData?.isAdmin)
```

Comment:
This is getting rather complex, maybe we should have a useMemo or useCallback somewhere in the beggining of this component (and even create an util if used elsewhere?)

Reply:
I don't understand how a useCallback can help so I just created a variable